sdk/provisioning/Azure.Provisioning.CloudMachine/src/CDKLevel3/CloudMachineInfrastructure.cs

```diff
@@ -230,12 +233,19 @@ public ProvisioningPlan Build(ProvisioningContext? context = null)
         _infrastructure.Add(_serviceBusSubscription_private);
         _infrastructure.Add(_serviceBusSubscription_default);
 
-        RoleAssignment roleAssignment = _serviceBusNamespace.CreateRoleAssignment(ServiceBusBuiltInRole.AzureServiceBusDataSender, RoleManagementPrincipalType.User, PrincipalIdParameter);
+        // This is necessary until SystemTopic adds an AssignRole method.
+        var role = ServiceBusBuiltInRole.AzureServiceBusDataSender;
+        RoleAssignment roleAssignment = new RoleAssignment("cm_servicebus_role");
+        roleAssignment.Name = BicepFunction.CreateGuid(_serviceBusNamespace.Id, Identity.Id, BicepFunction.GetSubscriptionResourceId("Microsoft.Authorization/roleDefinitions", role.ToString()));
+        roleAssignment.Scope = new IdentifierExpression(_serviceBusNamespace.IdentifierName);
+        roleAssignment.PrincipalType = RoleManagementPrincipalType.ServicePrincipal;
+        roleAssignment.RoleDefinitionId = BicepFunction.GetSubscriptionResourceId("Microsoft.Authorization/roleDefinitions", role.ToString());
+        roleAssignment.PrincipalId = Identity.PrincipalId;
```

Comment:
Where does `SystemTopic` factor in here?  Is there something you need added to `CreateRoleAssignment` to make this work?

Reply:
Yeah, i think there is an issue here, but it's code from Chris, and so I will let him deal with it